pkg/controller/controller.go

```diff
@@ -145,6 +164,26 @@ func (c *controller) brokerDelete(obj interface{}) {
 	glog.V(4).Infof("Received delete event for Broker %v", broker.Name)
 }
 
+// worker runs a worker thread that just dequeues items, processes them, and marks them done.
+// It enforces that the reconcileBrokerKey is never invoked concurrently with the same key.
+func (c *controller) brokerWorker() {
+	for {
+		func() {
```

Comment:
Does this func need to be here? It wraps something that is immediately called. Is it so the defer gets run? Should we just put the Done at the end? I don't see any other way out of the loop.